utils/build-script-impl

```diff
@@ -2076,14 +2076,19 @@ for host in "${ALL_HOSTS[@]}"; do
             libdispatch)
                 LIBDISPATCH_BUILD_DIR=$(build_directory ${host} ${product})
                 SWIFT_BUILD_PATH="$(build_directory ${host} swift)"
+                SWIFTC_BIN="$(build_directory_bin ${LOCAL_HOST} swift)/swiftc"
+                LLVM_BIN="$(build_directory_bin ${LOCAL_HOST} llvm)"
 
                 if [[ ! -f "${LIBDISPATCH_BUILD_DIR}"/config.status ]]; then
                     # First time building; need to run autotools and configure
                     call mkdir -p "${LIBDISPATCH_BUILD_DIR}"
                     with_pushd "${LIBDISPATCH_SOURCE_DIR}" \
                         call autoreconf -fvi
                     with_pushd "${LIBDISPATCH_BUILD_DIR}" \
-                        call "${LIBDISPATCH_SOURCE_DIR}"/configure --prefix="$(get_host_install_destdir ${host})$(get_host_install_prefix ${host})" --with-swift-toolchain="${SWIFT_BUILD_PATH}"
+                               call env CC="${LLVM_BIN}/clang" CCX="${LLVM_BIN}/clang" SWIFTC="${SWIFTC_BIN}" \
+                               "${LIBDISPATCH_SOURCE_DIR}"/configure --with-swift-toolchain="${SWIFT_BUILD_PATH}" \
+                               --prefix="$(get_host_install_destdir ${host})$(get_host_install_prefix ${host})"
```

Comment:
is libdispatch looking for executables in `SWIFT_BUILD_PATH` (i.e. `--with-swift-toolchain`) or just the standard library? If executables, it should switch to using `$(build_directory ${LOCAL_HOST} swift)`. I didn't know how this was used when factoring in cross-compiling support, but it seems like something we could fix here if it's not already correct. We might not be able to execute things from `SWIFT_BUILD_PATH`, but we should set the swift compiler's `-resource-dir` to `$(build_directory ${host} swift)/lib/swift` at any rate.